### ISSUE TITLE
No module named 'custom_conv'

### ISSUE THREAD

**Issue by stefenmax** [teboli/CPCR#3]
When i used   setup.py it raised:
File "setup.py", line 26, in <module>
    import custom_conv
ModuleNotFoundError: No module named 'custom_conv'
how can i fix it thanks!

**Comment by teboli** [comment]
Hi! This means that the cuda module has not compiled on your machine. However, it is only necessary if you intend to use the non-uniform model. If not, you can skip running setup.py and comment https://github.com/teboli/CPCR/blob/master/layers/functions.py#L7 (import custom_conv in this case). 

I will strive to fix this bug ASAP, sorry for inconvenience.

**Comment by stefenmax** [comment]
OK Thanks, it works. But does it mean if the cuda has not compiled on my machine, the code can't using the GPU.

**Comment by stefenmax** [comment]
By the way, the compile is failed cause

subprocess.CalledProcessError: Command '['which', 'g++']' returned non-zero exit status 1.

**Comment by teboli** [comment]
> OK Thanks, it works. But does it mean if the cuda has not compiled on my machine, the code can't using the GPU.

This is true only for my custon non-uniform convolution layer whose cuda code was shipped with the rest of the project. However compiling it is machine-dependent and I must think about a way to fix it. On the other hand, the code for uniform deblurring only rely on Pytorch implementation and thus is already compiled for your specific machine; you can run the method for uniform deblurring on your own GPU.
